vehicle/ICar.h

```diff
@@ -0,0 +1,29 @@
+#ifndef ICAR_H
+#define ICAR_H
+
+#include <string>
+#include <memory>
+#include "IEngine.h"
+#include "IGear.h"
+
+// Car interface
+class Car {
+public:
+    Car(std::unique_ptr<Engine> engine, std::unique_ptr<Gear> gear)
+        : engine(std::move(engine)), gear(std::move(gear)) {}
+
+    virtual std::string getCarType() const = 0;
+    const Engine* getEngine() const {
+        return engine.get();
+    }
+    const Gear* getGear() const {
+        return gear.get();
+    }
+    virtual ~Car() = default;
+
+protected:
+    std::unique_ptr<Engine> engine;
```

Comment:
note: since both of engine and gear are unlikely to be changed throughout the existence of a car object, you can simply create them on stack (but that's completely up to you)